bin/git-bulk

```diff
@@ -14,15 +14,24 @@ allwsmode=false
 #
 usage() {
   echo 1>&2 "usage: git bulk [-g] ([-a]|[-w <ws-name>]) <git command>"
-  echo 1>&2 "       git bulk --addworkspace <ws-name> <ws-root-directory>"
+  echo 1>&2 "       git bulk --addworkspace <ws-name> <ws-root-directory> (--from <URL or file>)"
   echo 1>&2 "       git bulk --removeworkspace <ws-name>"
   echo 1>&2 "       git bulk --addcurrent <ws-name>"
   echo 1>&2 "       git bulk --purge"
   echo 1>&2 "       git bulk --listall"
 }
 
 # add another workspace to global git config
-function addworkspace { git config --global bulkworkspaces."$wsname" "$wsdir"; }
+function addworkspace {
+	git config --global bulkworkspaces."$wsname" "$wsdir";
+	if [ ! -z "$source" ]; then
+		if ! cd "$wsdir" 2>/dev/null; then echo "Path of workspace doesn't exist, make it first."; exit 1; fi
+		if [[ "$source" =~ http?(s):// ]]; then
```

Comment:
Maybe we also need to support pattern like `git@github.com:org/repo`?

Reply:
I looked in [git docs](https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols) and tried to make a regex to match all of possible formats. But not sure whether if it is too complex and eventually a failure point or not…